unix_integration/src/resolver.rs

```diff
@@ -1047,6 +1062,10 @@ where
                             }
                         }
                     }
+                    _ => {
```

Comment:
We end up with weird matches here, such as (AuthCredHandler::Password, PamAuthRequest::DeviceAuthorizationGrant { .. }), (AuthCredHandler::DeviceAuthorizationGrant, PamAuthRequest::Password { cred }), which don't make any sense. That's why I did the wild card match. Should we always list all permutations? This could get rather large and ugly.

Reply:
You can just do:

```
(AuthCredHandler::Password, PamAuthRequest::DeviceAuthorisationGrant { .. } ) |
(AuthCredHandler::DeviceGrant { .. }, PamAuthRequest::Password { .. } ) => {
    ...
}
```

This allows multiple matches to asingle output here, and again, the point is to help guide future developers when we do make these changes. I'd rather it be correct and exhaustive, even if ugly, then have us forget to update something and consider possibilities.